### PR TITLE
improve unit test code coverage for storage.go to 77%

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
only missing function in storage.go is about initializing a database

overall percentage 26% -> 50%

Fixes 

## Type of change

Please delete options that are not relevant.

- Unit tests (no changes in the code)

## Testing steps

Please describe how the change was tested locally. If, for some reason, the testing was not done or not done fully, please describe what are the testing steps.

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by dpensi — APPROVED** [review]

File: storage.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func initDatabaseConnection(configuration StorageConfiguration) (*sql.DB, error)
 // multiple users have disabled some rules.
 func displayMultipleRuleDisable(connection *sql.DB, output string) error {
 	var fout *os.File = nil
-	var writer *bufio.Writer = nil
+	writer := bufio.NewWriter(os.Stdout)
```
Comment (tisnik) [inline storage.go:109]: it does not seem to be correct - it will output to stdout which is not always needed. + the defer might not work as well as expected.

File: storage.go [hunk 2]
```diff
@@ -265,7 +264,7 @@ func readOrgID(connection *sql.DB, clusterName string) (int, error) {
 // older than the specified time duration. Those records are simply displayed.
 func displayAllOldRecords(connection *sql.DB, maxAge, output string) error {
 	var fout *os.File = nil
-	var writer *bufio.Writer = nil
+	writer := bufio.NewWriter(os.Stdout)
```
Comment (tisnik) [inline storage.go:267]: dtto

File: storage_test.go [hunk 3]
```diff
@@ -241,6 +242,179 @@ func TestPerformDisplayMultipleRuleDisableOnError(t *testing.T) {
...
+	query1 := `
+                select cluster_id, rule_id, count(*) as cnt
+                  from cluster_rule_toggle
+                 group by cluster_id, rule_id
+                having count(*)>1
+                 order by cnt desc;
+`
+	// call the tested function
+	err = cleaner.PerformDisplayMultipleRuleDisable(connection, nil, query1, "cluster_rule_toggle")
+	if err != nil {
```
Comment (tisnik) [inline storage_test.go:281]: `assert.Error` or `assert.NoError` is better

File: storage_test.go [hunk 4]
```diff
@@ -241,6 +242,179 @@ func TestPerformDisplayMultipleRuleDisableOnError(t *testing.T) {
...
+		t.Errorf("error was not expected while updating stats: %s", err)
+	}
+
+	err = connection.Close()
+	if err != nil {
```
Comment (tisnik) [inline storage_test.go:286]: dtto with assertion

File: storage_test.go (hunk 3 above)
Comment (tisnik) [reply storage_test.go:281]: the tricky part - check if the function does it's job, i.e. what is displayed actually?

File: storage_test.go [hunk 5]
```diff
@@ -435,3 +685,134 @@ func TestDeleteRecordFromTableOnError(t *testing.T) {
...
+		for _, tableAndKey := range cleaner.TablesAndKeys {
+			// expected query performed by tested function
+			expectedExec := fmt.Sprintf("DELETE FROM %v WHERE %v = \\$", tableAndKey.TableName, tableAndKey.KeyName)
+			mock.ExpectExec(expectedExec).WithArgs(clusterName).WillReturnResult(sqlmock.NewResult(1, 2))
+		}
+	}
+
+	mock.ExpectClose()
+
+	_, err = cleaner.PerformCleanupInDB(connection, clusterNames)
```
Comment (tisnik) [inline storage_test.go:803]: todo -check for actual return value?

**Review by tisnik — COMMENTED** [review]
It looks ok, just added couple of comment/ideas

**Review by tisnik — CHANGES_REQUESTED** [review]
Please add tests for negative cases too. It might improve coverage a bit :)

File: storage.go (hunk 1 above)
Comment (Bee-lee) [reply storage.go]: done

File: storage.go (hunk 2 above)
Comment (Bee-lee) [reply storage.go]: done

File: storage_test.go (hunk 3 above)
Comment (Bee-lee) [reply storage_test.go:331]: done both

File: storage_test.go (hunk 5 above)
Comment (Bee-lee) [reply storage_test.go]: done

**Comment by Bee-lee** [comment]
@tisnik added negative test cases for scan errors, added check for output from `PerformCleanupInDB`, added checks for generated files from `DisplayAllOldRecords` and `DisplayMultipleRuleDisable`, added asserts instead of err checks, DRYed up some stuff

**Review by tisnik — APPROVED** [review]
LGTM
